morpheus/service/milvus_vector_db_service.py

```diff
@@ -137,31 +239,38 @@ def create(self, name: str, overwrite: bool = False, **kwargs: dict[str, typing.
                 for part in partition_conf["partitions"]:
                     self._client.create_partition(collection_name=name, partition_name=part["name"], timeout=timeout)
 
-    @with_mutex("_mutex")
-    def insert(self, name: str, data: typing.Union[list[list], list[dict], dict], **kwargs: dict[str, typing.Any]):
+    @with_collection_lock
+    def insert(self, name: str, data: typing.Union[list[list], list[dict], dict], **kwargs: dict[str,
+                                                                                                 typing.Any]) -> dict:
         """
         Insert a collection specific data in the Milvus vector database.
 
         Parameters
         ----------
         name : str
             Name of the collection to be inserted.
-        data : typing.Any
+        data : typing.Union[list[list], list[dict], dict]
```

Comment:
```suggestion
        data : list[list] | list[dict] | dict
```
We can use this new syntax